onnx/common/tensor.h

```diff
@@ -139,6 +151,219 @@ struct Tensor final {
+    const Tensor* a = &a_tensor;                                               \
+    if (a->elem_type() != elem_type_) {                                        \
+      throw("Type of tensors do not match");                                   \
+    }                                                                          \
+    if (a->sizes() != sizes_) {                                                \
+      throw("Tensor shapes are incompatible");                                 \
+    }                                                                          \
+    switch(elem_type_) {                                                       \
+      case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:  {                      \
+        CALL_BIN_FUNC(float, floats, f)                                        \
+        break;                                                                 \
+      }                                                                        \
+      case ONNX_NAMESPACE::TensorProto_DataType_BOOL:                          \
+      case ONNX_NAMESPACE::TensorProto_DataType_INT8:                          \
+      case ONNX_NAMESPACE::TensorProto_DataType_INT16:                         \
+      case ONNX_NAMESPACE::TensorProto_DataType_INT32:                         \
+      case ONNX_NAMESPACE::TensorProto_DataType_UINT8:                         \
+      case ONNX_NAMESPACE::TensorProto_DataType_UINT16:  {                     \
+        CALL_BIN_FUNC(int32_t, int32s, f)                                      \
+        break;                                                                 \
+      }                                                                        \
+      case ONNX_NAMESPACE::TensorProto_DataType_INT64:  {                      \
+        CALL_BIN_FUNC(int64_t, int64s, f)                                      \
+        break;                                                                 \
+      }                                                                        \
+      case ONNX_NAMESPACE::TensorProto_DataType_UINT32:                        \
+      case ONNX_NAMESPACE::TensorProto_DataType_UINT64:  {                     \
+        CALL_BIN_FUNC(uint64_t, uint64s, f)                                    \
+        break;                                                                 \
+      }                                                                        \
+      case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE: {                      \
+        CALL_BIN_FUNC(double, doubles, f)                                      \
+        break;                                                                 \
+      }                                                                        \
+      default:                                                                 \
+        throw("Operation not supported for this data type");                   \
```

Comment:
put elem_type_ into the error message to make it clear

Reply:
fixed